src/Common/FitOnFhir.Common/Services/AuthStateService.cs

```diff
@@ -13,20 +15,37 @@
 using Microsoft.Health.FitOnFhir.Common.ExtensionMethods;
 using Microsoft.Health.FitOnFhir.Common.Interfaces;
 using Microsoft.Health.FitOnFhir.Common.Models;
+using Newtonsoft.Json;
 
 namespace Microsoft.Health.FitOnFhir.Common.Services
 {
     public class AuthStateService : IAuthStateService
     {
+        private string _base36Chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+        private Random _random = new Random();
+        private BlobContainerClient _blobContainerClient;
+        private BlobServiceClient _blobServiceClient;
         private readonly AuthenticationConfiguration _authenticationConfiguration;
         private readonly IJwtSecurityTokenHandlerProvider _jwtSecurityTokenHandlerProvider;
         private readonly ILogger _logger;
 
         public AuthStateService(
+            AzureConfiguration azureConfiguration,
             AuthenticationConfiguration authenticationConfiguration,
             IJwtSecurityTokenHandlerProvider jwtSecurityTokenHandlerProvider,
+            BlobServiceClient blobServiceClient,
             ILogger<AuthStateService> logger)
         {
+            if (blobServiceClient == null)
+            {
+                _blobServiceClient = new BlobServiceClient(azureConfiguration.StorageAccountConnectionString);
```

Comment:
nit: We probably want to check if the azureConfiguration.StorageAccountConnectionString is null,